### ISSUE TITLE
Floating point exception (core dumped) with onednn opt

### ISSUE THREAD

**Issue by Shuo-Sun20** [tensorflow/tensorflow#77131]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

tf 2.17

### Custom code

Yes

### OS platform and distribution

linux 

### Python version

3.10

### Current behavior?

I don't know why but tensorflow allows a layer generating tensor with shapes containing 0, which will lead to 'Floating point exception (core dumped)' when oneDNN Opt is on. 

In the code below a MaxPooling1D should have generate a tensor with shape [3,5,0], while the engine crashed due to the bug I mentioned above.

### Standalone code to reproduce the issue

```shell
https://colab.research.google.com/drive/1jkQOOOjU2B60Qe8MvaEISXObuHBxm-I9?usp=sharing
```

### Relevant log output

```shell
The colab engine crashed.
If running locally, the error msg is:

Floating point exception (core dumped)
```

**Comment by AdvaitDongre** [comment]
heyy, i think these are some suggestions from my side which could help, let me know which one would help you:
1. try updating the tensorflow to latest version
2. try disabling oneDNN optimization `TF_DISABLE_MKL=1`
3. try checking tensor shapes before pooling, just ensure they've valid dim and dosen't have zero's in it

**Comment by Shuo-Sun20** [comment]
@AdvaitDongre 
Actually, I've tried them.
1. I'v used the latest version tf 2.17
2. Disabling oneDNN can avoid this problem but makes the executing process slow.
3. I'm executing a random experiment, and the shape are not determined before execution.

I thought the layer API could throw an exception which can be caught by 'try-except' block (as in Pytorch), but this floating point exception just makes the whole program crash.

**Comment by AdvaitDongre** [comment]
ahh got it, 
1. can you try prechecking for zero dimensions? 
gpt code for it:
`def safe_layer(layer, input_tensor):
    shape = tf.shape(input_tensor)
    if tf.reduce_any(shape == 0):
        print(f"Skipping layer {layer.name} due to zero dimension in shape {shape}")
        return input_tensor
    try:
        return layer(input_tensor)
    except Exception as e:
        print(f"Error applying layer {layer.name}: {e}")
        return input_tensor`

2. you could also try using tf.debugging.assert 
docs: [https://www.tensorflow.org/api_docs/python/tf/debugging/Assert](tf.debugging.assert)

but i think pre-checking would fix your error, let me know whether it does or not

**Comment by tilakrayal** [comment]
@Shuo-Sun20,
Hi, By default the colab notebook is using tensorflow v2.17 which contains keras3.0 which was causing the error. Could you please try to import keras2.0 with the below commands.

```python
!pip install tf-keras

import tf_keras as keras
```

Also I have changed some steps and the code was executed without crash/fail. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/99396471aaa4cf5e4d54cdb58dd808df/untitled2155.ipynb).

Thank you!

**Comment by Shuo-Sun20** [comment]
@AdvaitDongre 
Thank you for your advice.
Unfortunately, this error can not be caught by 'try-except' block. It just crashes the whole program. Therefore the code you provided can not work.

@tilakrayal 
Thank you for the check.
It seems that something goes wrongly when keras updating from 2.0 to 3.0. I'll turn to keras and report this issue.
